libraries/raw-rs/src/postprocessing/gamma_correction.rs

```diff
@@ -0,0 +1,86 @@
+use crate::Image;
+use std::f64::consts::E;
+
+pub fn gamma_correction(mut image: Image<u16>) -> Image<u16> {
+	let Some(histogram) = image.histogram else { return image };
+
+	let percentage = image.width * image.height;
+
+	let mut white = 0;
+	for channel_histogram in histogram {
+		let mut total = 0;
+		for i in (0x20..0x2000).rev() {
+			total += channel_histogram[i] as u64;
+
+			if total * 100 > percentage as u64 {
+				white = white.max(i);
+				break;
+			}
+		}
+	}
+
+	let curve = generate_gamma_curve(0.45, 4.5, (white << 3) as f64);
+
+	for value in image.data.iter_mut() {
+		*value = curve[*value as usize];
+	}
+	image.histogram = None;
+
+	image
+}
+
+fn generate_gamma_curve(power: f64, threshold: f64, max_intensity: f64) -> Vec<u16> {
+	let (mut bound_start, mut bound_end) = if threshold >= 1. { (0., 1.) } else { (1., 0.) };
+
+	let mut transition_point = 0.;
+	let mut transition_ratio = 0.;
+	let mut curve_adjustment = 0.;
+
+	if threshold != 0. && (threshold - 1.) * (power - 1.) <= 0. {
+		for _ in 0..48 {
+			transition_point = (bound_start + bound_end) / 2.;
+
+			if power != 0. {
+				let portion_of_threshold = transition_point / threshold;
+				let exponential_power = portion_of_threshold.powf(-power);
+				let power_ratio = (exponential_power - 1.) / (power - 1.); // TODO: guarantee that `power` cannot be 1
```

Comment:
Please address this and the following two TODO comments I added about division by zero being possible. I checked over all the other divisions and I'm confident they can't cause a panic, but these three probably can.

Reply:
Actually looking at the code once again it seems I messed up the order of operations here. I am frankly surprised it resulted in an almost correct image. The same mistake was made below too. I will have to rewrite the code.

As for `max_intensity`, the value of `max_intensity` comes from `white`. `white` is the maximum of `total` variable for all three channel histograms. `total` is incremented until it reaches 1% of the image area. So until the image itself is very small (less than 100 pixels), `max_intensity` should not be zero.

Reply:
Thanks for the fix.

![capture](https://github.com/user-attachments/assets/4aea75b3-a34e-4fcd-bdd1-5ee82c1b169a)

This fix you made was actually undoing my mistake, thank you for catching that. I changed that line during my code review, erroneously thinking it was equivalent. The others, I'm pretty sure, weren't my doing so I'm glad it ultimately got caught too.